### PR TITLE
fix(gatsby): Fix various small DEV_SSR bugs exposed in development_runtime tests

### PR DESCRIPTION
**Pull request by KyleAMathews**
PR includes some misc test fixes to fixes issues in tests.

### CODE REVIEW

File: e2e-tests/development-runtime/cypress/integration/functionality/hooks.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ const COUNT_ID = `count`
 
 describe(`hooks`, () => {
   beforeEach(() => {
-    cy.visit(`/hooks`).waitForRouteChange()
+    cy.visit(`/hooks`, { failOnStatusCode: false }).waitForRouteChange()
```
Comment (KyleAMathews) [inline e2e-tests/development-runtime/cypress/integration/functionality/hooks.js:5]: The dev server returns 404s now so don't fail tests

File: packages/gatsby/cache-dir/app.js [hunk 2]
```diff
@@ -126,7 +126,7 @@ apiRunnerAsync(`onClientEntry`).then(() => {
     undefined,
     // Client only pages have any empty body so we just do a normal
     // render to avoid React complaining about hydration mis-matches.
-    document.getElementById(`___gatsby`).children.length === 0
+    document.getElementById(`gatsby-focus-wrapper`).children.length === 0
```
Comment (KyleAMathews) [inline packages/gatsby/cache-dir/app.js:129]: `___gatsby` always has a child so wasn't actually testing if there was SSRed content or not so client-only pages would show a hydration warning.

File: packages/gatsby/src/utils/dev-ssr/develop-html-route.ts [hunk 3]
```diff
@@ -10,13 +10,13 @@ export const route = ({ app, program, store }): any =>
   app.get(`*`, async (req, res, next) => {
     trackFeatureIsUsed(`GATSBY_EXPERIMENTAL_DEV_SSR`)
 
-    const pathObj = findPageByPath(store.getState(), req.path)
+    const pathObj = findPageByPath(store.getState(), decodeURI(req.path))
```
Comment (KyleAMathews) [inline packages/gatsby/src/utils/dev-ssr/develop-html-route.ts:13]: Handle paths with unicode characters

File: packages/gatsby/src/utils/dev-ssr/render-dev-html.ts [hunk 4]
```diff
@@ -59,8 +59,13 @@ export const restartWorker = (htmlComponentRendererPath): void => {
 
 const searchFileForString = (substring, filePath): Promise<boolean> =>
   new Promise(resolve => {
+    const escapedSubString = substring.replace(/[.*+?^${}()|[\]\\]/g, `\\$&`)
```
Comment (KyleAMathews) [inline packages/gatsby/src/utils/dev-ssr/render-dev-html.ts:62]: Handle paths with special characters

**Review by pieh — APPROVED** [review]

**Pull request merged**